File: vortexdbg-api/src/main/java/com/vortexdbg/arm/NestedRun.java
package com.vortexdbg.arm;

import com.vortexdbg.LongJumpException;

public class NestedRun extends LongJumpException {

    public final long pc;

    private NestedRun(long pc) {
        this.pc = pc;
    }

    /**
     * need custom fix call context.
     */
    public static NestedRun runToFunction(long pc) {
        return new NestedRun(pc);
    }

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/arm/NestedRun.kt
package com.vortexdbg.arm

import com.vortexdbg.LongJumpException

class NestedRun private constructor(@JvmField val pc: Long) : LongJumpException() {

    companion object {
        /**
         * Requests that emulation resume at [pc] instead of returning from the hook;
         * the caller is responsible for setting up the call context (arguments, LR).
         */
        @JvmStatic
        fun runToFunction(pc: Long): NestedRun {
            return NestedRun(pc)
        }
    }

}
